src/mpi/romio/adio/common/strfns.c

```diff
@@ -102,3 +102,18 @@ char *ADIOI_Strdup(const char *str)
     }
     return save_p;
 }
+
+/*
+ * Convert string to lowercase
+ * Converts inplace, returns pointer to str
+ */
+char *ADIOI_Strlower(char *str)
```

Comment:
We use both `strcasecmp` and `strncasecmp` in the rest of MPICH unguarded. An MPL version that uses `tolower` in case no `str[n]casecmp` might be a good thing to have.